runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java

```diff
@@ -81,7 +81,10 @@ private void onClientShutdown(ClientShutdown e)
 		{
 			try
 			{
-				sessionClient.delete(sessionId);
+				if (sessionId != null)

```

Comment:
There are potentially ~3 threads here (event, which comes in on EDT?), this thread, and the thread executing `ping`, so this is not thread safe. You should make a local reference to sessionId and then use that.

Reply:
Good catch, I hadn't thought of the threaded nature of this.